kotlinx-coroutines-core/common/src/CancellableContinuationImpl.kt

```diff
@@ -119,12 +163,21 @@ internal open class CancellableContinuationImpl<in T>(
             // Invoke cancel handler if it was present
             if (state is CancelHandler) invokeHandlerSafely { state.invoke(cause) }
             // Complete state update
-            disposeParentHandle()
+            detachChildIfNonResuable()
             dispatchResume(mode = MODE_ATOMIC_DEFAULT)
             return true
         }
     }
 
+    internal fun parentCancelled(cause: Throwable) {
+        /*
+         * Here we can't reliably say whether postponed cancellation will be successful, but as it's internal API
+         * and we do not rely on return value, we are free to return `true`.
```

Comment:
The comment is now outdated. `parentCancelled` does not need to return anything.